src/screens/SignIn/handler.ts

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useContext } from 'react'
+
+import { AxiosResponse } from 'axios'
+
+import AuthAdapter from 'adapters/auth'
+import * as Constants from 'constants/auth'
+import { AuthContext } from 'contexts/auth'
+import { allFieldsFilled } from 'helpers/formValidation'
+
+type SubmitHandler = {
+  requestSuccess: boolean
+  error: string
+  handleSubmit: (e: React.SyntheticEvent) => void
+}
+
+const SubmitHandler = (): SubmitHandler => {
+  const [requestSuccess, setRequestSuccess] = useState(false)
+  const [error, setError] = useState('')
+  const { dispatch } = useContext(AuthContext)
+
+  const handleSubmit = async (e: React.SyntheticEvent) => {
+    e.preventDefault()
+
+    const target = e.target as typeof e.target & {
+      email: { value: string }
+      password: { value: string }
+    }
+
+    // TODO: Add UI test for this scenario
+    if (!allFieldsFilled([target.email.value, target.password.value])) {
+      setError('Email and password are required')
+      return
+    }
+
+    await AuthAdapter.signIn(target.email.value, target.password.value)
+      .then(function (response: AxiosResponse) {
+        if (response.status === 200) {
+          dispatch({
+            type: Constants.AUTH,
+            payload: response.data.data
+          })
+          setRequestSuccess(true)
+        }
+      })
+      .catch(function (err) {
+        if (err.response.status === 400) {
+          setError('Invalid email or password')
+        } else {
+          setError('Something went wrong. Please try again')
+        }
+      })
+  }
+
+  return {
+    requestSuccess: requestSuccess,
```

Comment:
One last comment, why is it needed to make a key-value thing here?

Reply:
You're right, there is no need to be key-value pair. Fixed in 3a460d28c8e58ce062b783bde977b578c3de54df